configuration/scripts/ratos-logging.sh

```
@@ -0,0 +1,238 @@
+        log_error "$script_name completed with errors (exit code: $exit_code)" "script_lifecycle" "SCRIPT_ERROR"
+    fi
+}
+
+# Function to create a summary of the current log session
+create_log_summary() {
+    local script_name="$1"
+    local start_time="$2"
+    local end_time="${3:-$(get_timestamp)}"
+    
+    # Count log entries by level for this session
+    local error_count=$(grep -c '"level":50' "$RATOS_LOG_FILE" 2>/dev/null || echo 0)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)